src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,226 @@
+                    if (sentences[1].equals("1")){
+                        isDone=true;
+                    }
+                    tasks[sum].setDone(isDone);
+                    sum++;
+                }
+                else if (sentences[0].equals("E")){
+                    tasks[sum]=new Event(sentences[2],sentences[3]);
+                    if (sentences[1].equals("1")){
+                        isDone=true;
+                    }
+                    tasks[sum].setDone(isDone);
+                    sum++;
+                }
+            }
+            return sum;
+        }
+        catch (FileNotFoundException e){
+            System.out.println("\n\tThe file does not exist.\n");
+        }
+        return 0;
+    }
+
+    public static void writeFile(Task[] tasks, int sum){
+        try {
+            File f = new File("/Users/wendyw/Desktop/study/cz2006(cs2113)/ip/data/duke.txt");
+            clearFile();
+            FileWriter fw = new FileWriter(f);
+            for (int i = 0; i < sum; i++) {
+                fw.write(tasks[i].writeToFile()+"\n");
+            }
+            fw.close();
+        }
+        catch (IOException e){
+            System.out.println("\n\tTrouble with IO stream.\n");
+        }
+    }
+
+    public static void clearFile(){
+        try {
+            File f = new File("/Users/wendyw/Desktop/study/cz2006(cs2113)/ip/data/duke.txt");
+            FileWriter fw = new FileWriter(f);
+            fw.write("");
+            fw.flush();
+            fw.close();
+        }
+        catch (IOException e){
+            System.out.println("\n\tTrouble with IO stream.\n");
+        }
+    }
+
+
+    public static void main(String[] args) {
+
+        showWelcomeScreen();
+        int inputCount=0;
+        int outputCount=1;
+        String inputString;
+        String input;
+
+        int sumList=readFile(taskArray);
+
+        do {
+            Scanner in = new Scanner(System.in);
+            inputString = in.nextLine();
+            input = inputString.trim();
+
+            try {
+                if(input.equals("done") || input.equals("todo") || input.equals("event") || input.equals("deadline")){
+                    throw new emptyException();
+                }
+            } catch (DukeException e) {
+                continue;
+            }
+
+                if (inputString.equals("list")) {
+                    if (taskArray[0]==null) {
+                        System.out.println("List is empty.");
+                    }
+                    else {
+                        System.out.println("Here are the tasks in your list: ");
+                        while (outputCount<=inputCount) {
+                            System.out.println(taskArray[outputCount-1]);
+                            outputCount++;
+                        }
+                    }
+                    outputCount=1;
+                }
+
+                else if (inputString.contains("done ")) {

```

Comment:
"done", "todo", "deadline",....you may want to define them as constants. Also, you may want to abstract the conditionals out into methods, so that the main body is cleaner and more readable.